tests/test_integration/test_utils/test_aggregators.py

```diff
@@ -0,0 +1,362 @@
+    table_name = "toolkit_aggregators_test_table_sequences"
+    rows = [
+        RowWrite(
+            key=f"sequence_00{i}",
+            columns={
+                "externalId": f"sequence_00{i}",
+                "name": f"Sequence 00{i}",
+                "assetId": asset_list[i % len(asset_list)].id,  # Assign to one of the assets
+                "columns": [
+                    {"name": f"Column {j}", "valueType": "STRING", "externalId": f"column_{j}"} for j in range(1, 4)
+                ],
+            },
+        )
+        for i in range(1, SEQUENCE_COUNT + 1)  # +1 for inclusive range
+    ]
+    create_raw_table_with_data(
+        toolkit_client,
+        RawTable(db_name=raw_db, table_name=table_name),
+        rows,
+    )
+    transformation = TransformationWrite(
+        external_id=SEQUENCE_TRANSFORMATION,
+        name="Toolkit Aggregators Test Sequence Transformation",
+        destination=TransformationDestination("sequences"),
+        query=f"""SELECT externalId as externalId, name as name, assetId as assetId, columns as columns, dataset_id('{SEQUENCE_DATASET}') as dataSetId
+FROM `{raw_db}`.`{table_name}`""",
+        ignore_null_fields=True,
+    )
+    created = upsert_transformation_with_run(toolkit_client, transformation)
+    return created
+
+
+class TestAggregators:
+    aggregators: tuple[tuple[type[AssetCentricAggregator], str, str, int], ...] = (
+        (AssetAggregator, ASSET_TRANSFORMATION, ASSET_DATASET, ASSET_COUNT),
+        (TimeSeriesAggregator, TIMESERIES_TRANSFORMATION, TIMESERIES_DATASET, TIMESERIES_COUNT),
+        (EventAggregator, EVENT_TRANSFORMATION, EVENT_DATASET, EVENT_COUNT),
+        (FileAggregator, FILE_TRANSFORMATION, FILE_DATASET, FILE_COUNT),
+        (SequenceAggregator, SEQUENCE_TRANSFORMATION, SEQUENCE_DATASET, SEQUENCE_COUNT),
+    )
+
+    @pytest.mark.usefixtures("assets", "events", "files", "time_series", "sequences")
+    @pytest.mark.parametrize(
+        "aggregator_class, expected_transformation_external_id, expected_dataset_external_id, expected_count",
+        aggregators,
+    )
+    def test_aggregations(
+        self,
+        toolkit_client: ToolkitClient,
+        aggregator_class: type[AssetCentricAggregator],
+        expected_transformation_external_id: str,
+        expected_dataset_external_id: str,
+        expected_count: int,
+        root_asset: Asset,
+    ) -> None:
+        root = root_asset.external_id
+        aggregator = aggregator_class(toolkit_client)
+
+        actual_count = aggregator.count(root)
+        assert actual_count == expected_count
+
+        used_data_sets = aggregator.used_data_sets(root)
+        assert used_data_sets == [expected_dataset_external_id]
+
+        transformation_count = aggregator.transformation_count()
+        assert transformation_count >= 1  # We know at least one transformation is writing to the resource type.
+        used_transformations = aggregator.used_transformations(used_data_sets)
```

Comment:
test used_transformations with id as well as external id?

Reply:
I updated the TimeSeries transformation to use the `datasetId` instead of `dataset_id("{{external_id}})`.